### PR TITLE
Rework the export to use CLI options and property mappers

### LINKED ISSUE #13613: Modify `ExportUtils.exportRealm` to return `OutputStream`
Exporting a realm should not produce a representation / JSON but an `OutputStream` instead.

The JSON realm representation is valid for the legacy store. The new store has to deal with either this representation or a native file with (likely) a file-store based structure.

This task is to prepare API in `ExportUtils` so that the `exportRealm` methods return an `OutputStream` and uses legacy / new store implementation logic to generate the stream accordingly. This might need implementing the `strip`ping of sensitive data in the partal export logic, which is a change opposed to the current state, likely by introducing a new flag in `ExportOptions`: https://github.com/keycloak/keycloak/blob/ec808d28bb224fc243edd334aa549a8cf8d94ca7/services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java#L1078

This implies relevant changes in all export endpoints.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #13613

### CODE REVIEW

**Comment by ahus1** [comment]
@hmlnarik - this PR contains the draft for our meeting later today.

**Comment by ahus1** [comment]
@hmlnarik - this PR is now green and contains a test `ExportModelTest.java` which runs the two exporters for CHM as part of the model tests. 

I scheduled a meeting with the CND team about the CLI options and how and if to use the property mappers. It it turns out to me more difficult than expected to have the CLI mended, I might consider rolling back the Quarkus related changes. That way we would still be able to run the exports in the model tests for the upcoming file store.

**Comment by ahus1** [comment]
@pedroigor / @mabartos / @vmuzikar: As discussed in today's call, I explored how to use the `Picocli` to dynamically add CLI options for the export command. To me this looks good and I'd like to continue on that route.

My latest commit f63922c5d9aaaec111ba93d6dc66655b3d58fa0c contains the changes, the build is still running. 

I now see also the documentation to fall in the right place: The new export options start to appear in the "All Configurations" section. Once the import has been refactored in a similar way, I can start listing the options as "related configurations" in the server docs like we do on other pages already.

Once you give me a first indication that this is the way we want to continue, I'll squash all previous commits and polish this PR for the final review and merge. 

The export now works expected, and the output for `export --help` now also looks nice. 

```
Usage:

kc.sh export [OPTIONS]

Export data from realms to a file or directory.

Options:

--dir <path>         Set the path to a directory where files will be read from when importing or
                       created with the exported data.
--file <path>        Set the path to a file that will be read when importing or created with the
                       exported data.
-h, --help           This help message.

Export:

--realm <realm>      Set the name of the realm to export. If not set, all realms are going to be
                       exported.
--users <strategy>   Set how users should be exported. Possible values are: skip, realm_file,
                       same_file, different_files. Default: different_files.
--users-per-file <number>
                     Set the number of users per file. It is used only if 'users' is set to
                       'different_files'. Default: 50.
```

File: quarkus/config-api/src/main/java/org/keycloak/config/ExportOptions.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            .buildTime(false)
+            .build();
+
+    public static final Option<String> REALM = new OptionBuilder<>("realm", String.class)
+            .category(OptionCategory.EXPORT)
+            .description("Set the name of the realm to export. If not set, all realms are going to be exported.")
+            .buildTime(false)
+            .build();
+
+    public static final Option<String> USERS_PER_FILE = new OptionBuilder<>("users-per-file", String.class)
```
Comment (mabartos) [inline quarkus/config-api/src/main/java/org/keycloak/config/ExportOptions.java:40]: Could be an Integer, right?

Comment (ahus1) [reply quarkus/config-api/src/main/java/org/keycloak/config/ExportOptions.java]: Sure, this is now fixed.

**Comment by ahus1** [comment]
@hmlnarik - this is now ready for review. Thanks!

**Review by pedroigor — COMMENTED** [review]
@ahus1 Does it make sense to also change `import` to make it consistent with the changes herein introduced? We can probably remove a bit more code too and make it simpler.

**Comment by ahus1** [comment]
@pedroigor - yes, `import` will be the next step. But it would involve changing the factories and it will not only be about the command. This will be the next PR after this one. This PR has grown into more areas already, and I'd rather get it merged and then apply the agreed concepts also to the import.

**Review by pedroigor — APPROVED** [review]
@ahus1 We don't have tests for export help message :) We can improve later.

**Review by mabartos — APPROVED** [review]
@ahus1 LGTM; from Quarkus distro side.

**Review by hmlnarik — APPROVED** [review]
Thank you @ahus1! This is a great step towards zip exports.

**Pull request merged**